Sources/SwiftDoc/Module.swift

```diff
@@ -29,12 +29,22 @@ public final class Module {
                     fileManager.isReadableFile(atPath: url.path),
                     fileManager.fileExists(atPath: url.path, isDirectory: &isDirectory)
                 else {
-                    // Skip top-level Tests directory
-                    if isDirectory.boolValue == true,
-                       url.lastPathComponent == "Tests",
-                       directory.appendingPathComponent("Tests").path == url.path
-                    {
-                        directoryEnumerator.skipDescendants()
+                    if isDirectory.boolValue == true {
```

Comment:
I'm afraid this won't work. There is a logic bug, which was introduced in #229 and I overlooked it in the code review 🙈:

In line 28: `url.pathExtension == "swift"` is false, because of this all following conditions in the guard statement never get executed. Because of this, `isDirectory` is never updated and is always false, so the directories are not skipped.

To be honest, I find the logic for collection all files very hard to understand by now. Maybe we could make it less complicated and more verbose and go with something like this:
```swift
        for path in paths {
            let directory = URL(fileURLWithPath: path)
            guard let directoryEnumerator = fileManager.enumerator(at: directory, includingPropertiesForKeys: [.isDirectoryKey], options: [.skipsHiddenFiles, .skipsPackageDescendants]) else { continue }
            let ignoredDirectories = Set([
               directory.appendingPathComponent("node_modules"),
               directory.appendingPathComponent("Packages"),
               directory.appendingPathComponent("Pods"),
               directory.appendingPathComponent("Resources"),
               directory.appendingPathComponent("Tests"),
            ])
            for case let url as URL in directoryEnumerator {
                guard url.pathExtension == "swift", fileManager.isReadableFile(atPath: url.path)
                else {
                    if ignoredDirectories.contains(url) {
                        directoryEnumerator.skipDescendants()
                    }

                    continue
                }

                sources.append((url, directory))
            }
        }
```
(We could even

Reply:
Thanks for catching that. I like your approach, and followed that with dcdfe9716cc22a3373a19c74f0f224195f5d8bbc.